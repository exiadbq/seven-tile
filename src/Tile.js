import React from 'react';
import styled from 'styled-components';
import tile from './assets/tile.jpg';
import logo from './assets/logo.png';

const Wrapper = styled.div`
  position: relative;
  height: 100%;
  overflow: hidden;
  display: flex;
  justify-content: center;
  width: 100%;
  max-width: 40em;
  margin: auto;

  .overlay {
    position: absolute;
    height: 30%;
    bottom: 0;
    left: 50%;
    transform: translateX(-50%);
    width: 100%;
    background: rgba(0, 0, 0, 0.5);
    display: flex;
    align-items: center;

    img {
      background: transparent;
      height: 100%;
      padding: 0.5em;
    }
  }
`;
const Image = styled.img`
  width: 100%;
  position: relative;
`;

const H1 = styled.h1`
  color: ${props => props.color};
`;
const Tile = ({title}) => {
  return (
    <Wrapper>
      <Image src={tile} alt={title} />
      <div className="overlay">
        <img src={logo} alt={'seven'} />
        <H1 color="white">{title}</H1>
      </div>
    </Wrapper>
  );
};
export default Tile;
