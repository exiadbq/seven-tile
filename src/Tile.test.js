import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Tile from './Tile';

test('it works', () => {
	const tree = renderer.create(<Tile title={'test'}/>).toJSON()
	expect(tree).toMatchSnapshot()
});
