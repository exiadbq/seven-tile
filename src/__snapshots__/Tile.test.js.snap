// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  position: relative;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  max-width: 40em;
  margin: auto;
}

.c0 .overlay {
  position: absolute;
  height: 30%;
  bottom: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 100%;
  background: rgba(0,0,0,0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .overlay img {
  background: transparent;
  height: 100%;
  padding: 0.5em;
}

.c1 {
  width: 100%;
  position: relative;
}

.c2 {
  color: white;
}

<div
  className="c0"
>
  <img
    alt="test"
    className="c1"
    src="tile.jpg"
  />
  <div
    className="overlay"
  >
    <img
      alt="seven"
      src="logo.png"
    />
    <h1
      className="c2"
      color="white"
    >
      test
    </h1>
  </div>
</div>
`;
